src/Data/Validation.hs

```diff
@@ -660,14 +675,12 @@ optional (Just a) f =
     DisputedCtx gfs lfs b -> DisputedCtx gfs lfs (Just b)
     RefutedCtx gfs lfs    -> RefutedCtx gfs lfs
 
--- | Runs a set of validations only when a requirement is met.
-validateWhen :: Bool -> VCtx f b -> VCtx f (Maybe b)
-validateWhen False _ = pure Nothing
-validateWhen True v  = fmap Just v
-
--- | Runs a set of validations when a requirement is not met.
-validateUnless :: Bool -> VCtx f b -> VCtx f (Maybe b)
-validateUnless b = validateWhen (not b)
+-- | Allows for validation of an optional value.
+-- See `Validating Complex Types` for an example.
+whenJust :: (ValueCtx a -> VCtx f (ValueCtx b)) -> ValueCtx (Maybe a) -> VCtx f (ValueCtx (Maybe b))

```

Comment:
I wonder if `transformWhenJust` or something similar would be a little more explicit for this functions purpose.

Reply:
I think `whenJust` works much like the `when` and `unless` functions. The types and context clarify the rest.

Reply:
Fair enough, it mostly seems off to me because I work with same named functions in my codebase. Context is key though.